### PR TITLE
Update ISO URLs to latest, add script to update ISO checksums

### PR DESCRIPTION
**Pull request by abhay-krishna**
* Updating a few ISO URLs pointing to images that got removed from the downloads mirror
* Bump some Ubuntu ISO URLs to the latest stable point releases
* Add script to update Ubuntu ISO URL checksums

### CODE REVIEW

File: images/capi/packer/ova/ubuntu-2204-efi.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   "guest_os_type": "ubuntu-64",
   "iso_checksum": "10f19c5b2b8d6db711582e0e27f5116296c34fe4b313ba45f9b201a5007056cb",
   "iso_checksum_type": "sha256",
-  "iso_url": "https://old-releases.ubuntu.com/releases/jammy/ubuntu-22.04.1-live-server-amd64.iso",
+  "iso_url": "https://old-releases.ubuntu.com/releases/jammy/ubuntu-22.04.3-live-server-amd64.iso",
```
Comment (abhay-krishna) [inline images/capi/packer/ova/ubuntu-2204-efi.json:14]: This is present, just updating to latest point release.

File: images/capi/packer/ova/ubuntu-2204.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "guest_os_type": "ubuntu-64",
   "iso_checksum": "10f19c5b2b8d6db711582e0e27f5116296c34fe4b313ba45f9b201a5007056cb",
   "iso_checksum_type": "sha256",
-  "iso_url": "https://old-releases.ubuntu.com/releases/jammy/ubuntu-22.04.1-live-server-amd64.iso",
+  "iso_url": "https://old-releases.ubuntu.com/releases/jammy/ubuntu-22.04.3-live-server-amd64.iso",
```
Comment (abhay-krishna) [inline images/capi/packer/ova/ubuntu-2204.json:13]: The previous image is present, just updating to latest point release.

File: images/capi/packer/proxmox/ubuntu-2204.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "distribution_version": "2204",
   "distro_name": "ubuntu",
   "iso_checksum": "sha256:5e38b55d57d94ff029719342357325ed3bda38fa80054f9330dc789cd2d43931",
-  "iso_url": "https://old-releases.ubuntu.com/releases/jammy/ubuntu-22.04.2-live-server-amd64.iso",
+  "iso_url": "https://old-releases.ubuntu.com/releases/jammy/ubuntu-22.04.3-live-server-amd64.iso",
```
Comment (abhay-krishna) [inline images/capi/packer/proxmox/ubuntu-2204.json:7]: The previous image is present, just updating to latest point release.

**Comment by AverageMarcus** [comment]
The `iso_checksums` will need updating too.

**Comment by AverageMarcus** [comment]
Possibly but I was struggling to follow the logs. @mboersma do you have any idea what's causing it to fail?

**Comment by mboersma** [comment]
```
Installed plugin github.com/hashicorp/ansible v1.1.1 in "/root/.packer.d/plugins/github.com/hashicorp/ansible/packer-plugin-ansible_v1.1.1_x5.0_linux_amd64"
rockylinux-8: SUCCESS
flatcar: SUCCESS
photon-3: SUCCESS
photon-4: SUCCESS
photon-5: SUCCESS
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:169","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 1h0m0s timeout","severity":"error","time":"2024-03-04T19:01:44Z"}
++ early_exit_handler
```

Are we seeing this everywhere, or just with this PR? I'm also having a hard time seeing the root problem by looking at the logs.

**Comment by abhay-krishna** [comment]
I compared the failure logs with a [successful build](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_image-builder/1420/pull-ova-all/1763472640170790912) from a [previous commit](https://github.com/kubernetes-sigs/image-builder/commit/25846fc365a9d848eac9edd4f13d5bb8203faf43) in this PR.

The successful build had the following timestamp log at the top
`Running build with timestamp 20240301T075343` => 7:53:43 AM on March 1, 2024

Then it printed out the SUCCESS logs for each OS (like above) and got to the govc install step (for VM cleanup) at
`--2024-03-01 08:15:31--  https://github.com/vmware/govmomi/releases/download/v0.30.5/govc_Linux_x86_64.tar.gz` => 08:15:31 AM on March 1, 2024

So it took only about 22 minutes from the start to the `govc` install step.

However in the failed job logs, these two timestamps are an hour apart.
`Running build with timestamp 20240304T180145` => 06:01:45 PM on March 3, 2024
`--2024-03-04 19:01:44--  https://github.com/vmware/govmomi/releases/download/v0.30.5/govc_Linux_x86_64.tar.gz` => 07:01:44 PM on March 3, 2024.

**Comment by abhay-krishna** [comment]
@mboersma I think I figured out why the builds are timing out. Looking at the logs in the Artifacts folder, the [ubuntu 20.04 build](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_image-builder/1420/pull-ova-all/1764712815068712960/artifacts/ubuntu-2004.log) seems to be stuck at `Waiting for SSH to become available`. I will revert my changes to the Ubuntu 20.04 OVA URLs to see if it gets fixed.

File: images/capi/packer/raw/raw-ubuntu-2004-efi.json [hunk 4]
```diff
@@ -6,9 +6,9 @@
   "distro_name": "ubuntu",
   "firmware": "OVMF.fd",
   "guest_os_type": "ubuntu-64",
-  "iso_checksum": "f11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2",
+  "iso_checksum": "5035be37a7e9abbdc09f0d257f3e33416c1a0fb322ba860d42d74aa75c3468d4",
   "iso_checksum_type": "sha256",
-  "iso_url": "http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04/release/ubuntu-20.04.1-legacy-server-amd64.iso",
+  "iso_url": "https://old-releases.ubuntu.com/releases/20.04/ubuntu-20.04.5-live-server-amd64.iso",
```
Comment (mboersma) [inline images/capi/packer/raw/raw-ubuntu-2004-efi.json:11]: Thanks for standardizing all these download URLs too.

**Review by mboersma — COMMENTED** [review]
This looks good to me. I've tried it out locally and the `make` targets seem to work.

File: images/capi/hack/update-iso-checksums.sh [hunk 5]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (abhay-krishna) [reply images/capi/hack/update-iso-checksums.sh]: Fixed!

File: images/capi/packer/raw/raw-ubuntu-2004-efi.json (hunk 4 above)
Comment (abhay-krishna) [reply images/capi/packer/raw/raw-ubuntu-2004-efi.json:11]: I was looking to standardize to the same endpoint for OVAs too but it failed waiting for SSH. That part needs investigation.

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by drew-viles — APPROVED** [review]
Looks great. Thanks!

/lgtm

**Comment by AverageMarcus** [comment]
/lgtm
/approve

🚀

**Pull request merged**